### PR TITLE
feat: add long running migrating benchmark

### LINKED ISSUE #16220: Set up an long running update/migration benchmark
**Description**

Several recent incidents indicated issues with migration or updates to minor versions.

We have Update IT, which covers the general update from the last minor to the newest/recent minor (or main). Furthermore, we have tests to verify rolling update, but all of these are only do one update from the last to the recent and with not much data involved. 

We need a benchmark that is sort of forever running. This benchmark should always updated on the newest minors and patch releases. Ideally with a mixed workload to cover more features. This would had helped to find issues, like #15804.

**Alternatives:**

 * Extend update IT, we can cover more versions in our tests
 * Set up only for releases, such benchmark and always start from last supported version (.0 version to cover cases where people not going to latest patches) migrated to be released version

**Context**:

 * Related [incident](https://docs.google.com/document/d/11-IfrzlYVkOAliD0i-4YvulghCMe5AewpywfLP6O_YA/edit) 
 * Related bugs https://github.com/camunda/zeebe/issues/15804
 *

### PR DESCRIPTION
**Pull request by rodrigo-lourenco-lopes**
## Description

This workflow reuses the zeebe-benchmark.yml workflow to update the existing long-running-migrating-benchmark with a mixed load when the latest release gets published.

## Related issues

closes https://github.com/camunda/zeebe/issues/16220

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: .github/workflows/zeebe-update-long-running-migrating-benchmark.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+jobs:
+  setup-mixed-benchmark:
+    name: Update Long Running Migrating Benchmark with mixed load
+    uses: ./.github/workflows/zeebe-benchmark.yml
+    secrets: inherit
+    with:
+      name: long-running-migrating-benchmark
+      cluster: zeebe-cluster
+      cluster-region: europe-west1-b
+      ref: main # we run the latest changes
```
Comment (oleschoenburg) [inline .github/workflows/zeebe-update-long-running-migrating-benchmark.yaml:15]: :x: That's not what we want though. This benchmark shouldn't run the latest code from main but always the newest patch release of the newest minor version. You need to query releases, determine the correct version and then use that version as the zeebe tag.

File: .github/workflows/zeebe-update-long-running-migrating-benchmark.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      benchmark-load: >
+        --set starter.rate=5
+        --set worker.replicas=1
+        --set timer.replicas=1
+        --set timer.rate=5
+        --set publisher.replicas=1
+        --set publisher.rate=5
+        --set camunda-platform.operate.enabled=true
+        --set camunda-platform.operate.image.repository=gcr.io/zeebe-io/operate
+        --set camunda-platform.operate.image.tag=${{ needs.benchmark-data.outputs.operate }}
```
Comment (oleschoenburg) [inline .github/workflows/zeebe-update-long-running-migrating-benchmark.yaml:26]: :x: That step doesn't exist. Looks like a left over from a previous state?

**Review by lenaschoenburg — CHANGES_REQUESTED** [review]
This doesn't work yet. Have you tested this somewhere? You could temporarily add a `pull_request` trigger while you are developing.

**Review by lenaschoenburg — APPROVED** [review]
Nice work :tada:

**Pull request merged**